server/gin_html_handlers.go

```diff
@@ -0,0 +1,285 @@
+package server
+
+import (
+	"context"
+	"fmt"
+	"net/http"
+
+	"github.com/gin-gonic/gin"
+
+	"github.com/joshprzybyszewski/cribbage/model"
+	"github.com/joshprzybyszewski/cribbage/server/persistence"
+)
+
+func handleIndex(c *gin.Context) {
+	c.HTML(
+		http.StatusOK,
+		"index.html",
+		gin.H{},
+	)
+}
+
+func handleGetUser(c *gin.Context) {
+	// read the username/displayname from the query params
+	// and redirect to /user/:username
+	username := c.Query(`username`)
+	name := c.Query(`displayName`)
+
+	fmt.Printf("username: %s, displayname: %s\n", username, name)
+
+	c.Redirect(http.StatusTemporaryRedirect, fmt.Sprintf(`/user/%s`, username))
+}
+
+func (cs *cribbageServer) handleGetUsername(c *gin.Context) {
+	ctx := context.Background()
+	// serve up a list of games this user is in
+	username := c.Param("username")
+	pID := model.PlayerID(username)
+
+	db, err := cs.dbFactory.New(ctx)
+	if err != nil {
+		c.String(http.StatusInternalServerError, `dbFactory.New() error: %s`, err)
+		return
+	}
+	defer db.Close()
+
+	p, err := getPlayer(ctx, db, pID)
+	if err != nil {
+		c.String(http.StatusInternalServerError, "Error: %s", err)
+		return
+	}
+
+	gameNames := make([]struct {
+		Desc       string
+		MyColor    string
+		RedScore   int
+		BlueScore  int
+		GreenScore int
+		GameID     model.GameID
+	}, 0, len(p.Games))
+	for gID, color := range p.Games {
+		g, err := getGame(ctx, db, gID)
+		if err != nil {
+			if err == persistence.ErrGameNotFound {
+				// the player knows about a game that's been deleted
+				continue
+			}
+			c.String(http.StatusInternalServerError, "Error getting game: %s", err)
+			return
+		}
+		gameDesc := `Against `
+		for _, p := range g.Players {
+			if p.ID == pID {
+				continue
+			}
+			// TODO support three and four player games
+			gameDesc += p.Name
+		}
+		gameDesc += ` `
+		gameNames = append(gameNames, struct {
+			Desc       string
+			MyColor    string
+			RedScore   int
+			BlueScore  int
+			GreenScore int
+			GameID     model.GameID
+		}{
+			Desc:       gameDesc,
+			GameID:     gID,
+			MyColor:    color.String(),
+			RedScore:   g.CurrentScores[model.Red],
+			BlueScore:  g.CurrentScores[model.Blue],
+			GreenScore: g.CurrentScores[model.Green],
+		})
+	}
+
+	c.HTML(
+		http.StatusOK,
+		"user.html",
+		gin.H{
+			"displayName": username,
+			"myID":        string(pID),
+			"games":       gameNames,
+		},
+	)
+}
+
+func (cs *cribbageServer) handleGetUsernameGame(c *gin.Context) { //nolint:gocyclo
+	// serve up this game for this user
+	pID := model.PlayerID(c.Param("username"))
```

Comment:
:cactus: 📖  ` instead of "

Reply:
took me a second to decipher your comment;)